### PR TITLE
add customizeable colors to link portal and title (fixes #2930)

### LINKED ISSUE #2930: link with red color
when create a link with js. l can't delete the red color in circle.
bug
thank you with much.
```js
    var entityEl = document.createElement('a-entity');
    entityEl.setAttribute('id',`link-${index}`);
    entityEl.setAttribute('link', {
      href: `chart.html?machineId=${item.name}`,
      title: item.name,
      color: getColor(item.status),
      image: 'src/img/sunrise.png',
    });
    entityEl.setAttribute('position', getLoaction(index));
    entityEl.setAttribute('text', {
      value: Math.round(item.data.temp).toString(),
      height: 5,
      width: 8,
      color: 'white',
      align: 'center'
    });
```

### PR DESCRIPTION
**Pull request by MichaelHazani**
**Description:**
Fixes #2930 + adds customisable title color prop

**Changes proposed:**
- Adds prop to change link portal **background** color (rather than the hardcoded red)
- Adds prop to change link **title** color
- Refactors link declaration schema to be accordingly specific: borderColor, innerColor and titleColor
- Updates docs accordingly

### CODE REVIEW

File: docs/components/link.md [hunk 1]
```diff
@@ -60,14 +60,16 @@ We also provide a link primitive with a different syntax:
 
 | Property            | Description                                                                                                                                  | Default Value |
 |---------------------|----------------------------------------------------------------------------------------------------------------------------------------------|---------------|
-| color               | Background color of the portal.                                                                                                              | white         |
+| borderColor         | Border color of the portal.                                                                                                                  | white         |
+| innerColor          | Inner (background) color of the portal.                                                                                                      | red           |
```
Comment (ngokevin) [inline docs/components/link.md]: Perhaps `backgroundColor`

File: src/components/link.js [hunk 2]
```diff
@@ -31,15 +33,17 @@ module.exports.Component = registerComponent('link', {
...
     el.setAttribute('material', 'strokeColor', strokeColor);
+    el.setAttribute('material', 'innerColor', innerColor);
     if (data.on !== oldData.on) { this.updateEventListener(); }
     if (data.visualAspectEnabled && oldData.peekMode !== undefined && data.peekMode !== oldData.peekMode) {
       this.updatePeekMode();
     }
     if (!data.image || oldData.image === data.image) { return; }
     el.setAttribute('material', 'pano',
-                    typeof data.image === 'string' ? data.image : data.image.src);
+      typeof data.image === 'string' ? data.image : data.image.src);
```
Comment (ngokevin) [inline src/components/link.js]: don't need to change this

File: src/components/link.js [hunk 3]
```diff
@@ -92,15 +96,15 @@ module.exports.Component = registerComponent('link', {
     semiSphereEl = this.semiSphereEl = this.semiSphereEl || document.createElement('a-entity');
 
     // Set Portal
-    el.setAttribute('geometry', {primitive: 'circle', radius: 1.0, segments: 64});
+    el.setAttribute('geometry', { primitive: 'circle', radius: 1.0, segments: 64 });
```
Comment (ngokevin) [inline src/components/link.js]: don't need to change this

File: src/components/link.js [hunk 4]
```diff
@@ -307,9 +311,10 @@ module.exports.Component = registerComponent('link', {
 /* eslint-disable */
 registerShader('portal', {
   schema: {
-    pano: {type: 'map', is: 'uniform'},
-    borderEnabled: {default: 1.0, type: 'int', is: 'uniform'},
-    strokeColor: {default: 'white', type: 'color', is: 'uniform'}
+    pano: { type: 'map', is: 'uniform' },
```
Comment (ngokevin) [inline src/components/link.js]: don't need to add spaces to objects

**Comment by ngokevin** [comment]
thanks!

**Comment by MichaelHazani** [comment]
@ngokevin thanks and sorry about the weird spaces - working through some linting issues in my local dev env. Won't happen next time;)

**Comment by ngokevin** [comment]
Nah don't care, just fixed it up to merge it quick

**Pull request merged**